integration-cli/docker_cli_build_test.go

```diff
@@ -6401,3 +6402,44 @@ CMD echo foo
 	out, _ := dockerCmd(c, "inspect", "--format", "{{ json .Config.Cmd }}", "build2")
 	c.Assert(strings.TrimSpace(out), checker.Equals, `["/bin/sh","-c","echo foo"]`)
 }
+
+func (s *DockerSuite) TestBuildIidFile(c *check.C) {
+	tmpDir, err := ioutil.TempDir("", "TestBuildIidFile")
+	if err != nil {
+		c.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+	tmpIidFile := filepath.Join(tmpDir, "iid")
+
+	cli.BuildCmd(c, "testbuildiidfile",
+		build.WithDockerfile(`FROM `+minimalBaseImage()+`
+ENV FOO FOO`),
+		cli.WithFlags("--iidfile", tmpIidFile))
+
+	id, err := ioutil.ReadFile(tmpIidFile)
+	c.Assert(err, check.IsNil)
+	d := digest.FromBytes(id)
+	c.Assert(d.Validate(), check.IsNil)
```

Comment:
Indeed, in fact thre was also a lurking bug which that exposed: `digest.FromBytes(id)` calculates a new hash over the string representation of the hash! I should have been using `digest.Parse(string(id))` and now do.